commerce-service/src/main/java/com/liferay/commerce/internal/upgrade/v5_0_0/CommerceOrderUpgradeProcess.java

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.commerce.internal.upgrade.v5_0_0;
+
+import com.liferay.commerce.model.impl.CommerceOrderImpl;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.dao.jdbc.DataAccess;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.StringBundler;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.Statement;
+
+/**
+ * @author Luca Pellizzon
+ */
+public class CommerceOrderUpgradeProcess extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+
+		// Add commerceCurrencyCode column
+
+		String newColumnName = "commerceCurrencyCode";
+
+		_addColumn(
+			CommerceOrderImpl.class, CommerceOrderImpl.TABLE_NAME,
+			newColumnName, "VARCHAR(75)");
+
+		// Convert commerceCurrencyId to commerceCurrencyCode
+
+		PreparedStatement ps = null;
+		ResultSet rs = null;
+		Statement s = null;
+
+		try {
+			ps = connection.prepareStatement(
+				"update CommerceOrder set commerceCurrencyCode = ? where " +
+					"commerceCurrencyId = ?");
+
+			s = connection.createStatement();
+
+			rs = s.executeQuery(
+				"select commerceCurrencyId, code_ from CommerceCurrency");
```

Comment:
This would be nicer as a try-with-resources, if you don't mind. It saves several lines, and prevents us from having to remember to clean up at the end.